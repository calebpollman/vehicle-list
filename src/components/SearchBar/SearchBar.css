.search-bar-container {
  background-color: var(--color-search-bar-background);
  display: flex;
  height: 6%;
  padding: 2%;
  width: 96%;
}

.search-bar-icon-container {
  align-items: center;
  display: flex;
  margin-left: auto;
}

.search-bar-icon-container:hover {
  cursor: pointer;
}