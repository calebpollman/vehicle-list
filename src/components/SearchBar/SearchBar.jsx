import React from 'react';
import PropTypes from 'prop-types';
import SearchBarInput from '../SearchBarInput/SearchBarInput';
import {CloseIcon, SearchIcon} from '../Icons/Icons';
import './SearchBar.css';

const SearchBar = ({searchActive, toggleSearchActive, updateSearchTerm}) => {
  
  let input;
  if (searchActive) input = <SearchBarInput updateSearchTerm={updateSearchTerm} />;

  return (
    <div className="search-bar-container">
      <div className="search-bar-input-container">
        {input}  
      </div>
      <div 
        className="search-bar-icon-container"
        onClick={event => {toggleSearchActive(event)}}
      >
        {searchActive ? <CloseIcon /> : <SearchIcon />}
      </div>
    </div>
  );
}

SearchBar.propTypes = {
  searchActive: PropTypes.bool.isRequired, 
  toggleSearchActive: PropTypes.func.isRequired, 
  updateSearchTerm: PropTypes.func.isRequired,
}
export default SearchBar;