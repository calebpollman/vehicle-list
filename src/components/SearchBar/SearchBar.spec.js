import React from 'react';
import {shallow} from 'enzyme';
import SearchBar from './SearchBar';

describe('SearchBar', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SearchBar searchActive />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('renders 2 <div />\'s', () => {
    expect(wrapper.find('div').length).toEqual(2);
  });
});