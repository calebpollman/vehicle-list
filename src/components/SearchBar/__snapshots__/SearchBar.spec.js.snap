// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    searchActive={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SearchBarInput
          updateSearchTerm={undefined}
/>,
        <div
          className="search-bar-icon-container"
          onClick={[Function]}
>
          <Unknown />
</div>,
      ],
      "className": "search-bar-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "updateSearchTerm": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Unknown />,
          "className": "search-bar-icon-container",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SearchBarInput
            updateSearchTerm={undefined}
/>,
          <div
            className="search-bar-icon-container"
            onClick={[Function]}
>
            <Unknown />
</div>,
        ],
        "className": "search-bar-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "updateSearchTerm": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Unknown />,
            "className": "search-bar-icon-container",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
